### ISSUE TITLE
Skript doesn't get entities in chunk related events

### ISSUE THREAD

**Issue by HC-Emro** [SkriptLang/Skript#5829]
### Skript/Server Version

```markdown
[Skript] Server Version: git-Paper-538 (MC: 1.19.4)
[Skript] Skript Version: 2.7.0-beta3 (skriptlang-github)
[Skript] Installed Skript Addons: None
[Skript] Installed dependencies: None
```

### Bug Description

Skript doesn't get entities in chunk related events

### Expected Behavior

To get entities in event-chunk

### Steps to Reproduce

```
on chunk enter:
  set {_c} to event-chunk
    loop all entities in {_c}: # Loop doesn't work because it doesn't find any entities but there are
      broadcast "%loop-entity%"
```
or
```
on chunk unload:
  set {_c} to event-chunk
  loop all entities of type armor stand in {_c}: # Loop doesn't work because it doesn't find any entities but there are
    kill loop-entity
```
or
```
on chunk unload:
  set {_c} to event-chunk
  broadcast "%all entities in {_c}%" # --> spamming <none> but there are some entities
```

### Errors or Screenshots

No errors in console

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by Fusezion** [comment]
add a `wait a tick` to the line before you get entities

**Comment by HC-Emro** [comment]
```
on chunk enter:
    broadcast "Is there any entities here?"
	set {_c} to event-chunk
	wait a tick
	loop all entities in {_c}:
		broadcast "Yes ! : %loop-entity%"
```
![image](https://github.com/SkriptLang/Skript/assets/56135028/e7232dd4-0b9d-4439-ace3-4455d1c7ff52)
still not working :/

**Comment by iOshawott** [comment]
Hmmmm...

code
```
on chunk unload:
	loop event-chunk's entities:
		broadcast "%loop-entity%"
```
worked on Skript 2.6, for now it's not working

BUT
when I added wait 1 tick

```
on chunk unload:
	wait 1 tick
	loop event-chunk's entities:
		broadcast "%loop-entity%"
```
it's working again

And I just tested - code

```
on chunk enter:
	broadcast "Is there any entities here?
	loop event-chunk's entities:
		broadcast "Yes ! : %loop-entity%"
```
works too

**Comment by HC-Emro** [comment]
Oh yes, it's working now! Thank you very much! I was on this for a while.

**Comment by HC-Emro** [comment]
I'm coming back to you because ultimately the wait 1 tick is not at all optimized when you have more than 10 players on the server.

**Comment by DelayedGaming** [comment]
You can reopen this issue and wait until someone found the root cause, as technically I don’t see why with and without a tick delay would cause such difference.

**Comment by Fusezion** [comment]
Correct, don't recall saying it was optimized. In general I highly doubt a chunk that consists of 98.3k blocks would be efficient for 10 players for over 983k blocks. As DelayedGaming said you can reopen this from what I checked we don't even register an `event-chunk` which means it's most likely converting the player entity to chunk at player this is likely the cause of why it didn't work as the player's location will be the old chunk

**Comment by HC-Emro** [comment]
My goal is to remove minecarts and armor stands from chunks that are unloading. And for this, there should not be a delay, it worked perfectly with previous versions of Skript, I hope that the developers can quickly resolve this problem.

**Comment by DelayedGaming** [comment]
I was referring to the entities expression returns nothing when a delay is not used. My bad for not clarifying.

**Comment by Moderocky** [comment]
> as technically I don’t see why with and without a tick delay would cause such difference.

Most events are broadcast before the thing actually happens (hence why you are able to cancel them, _id est_ prevent the thing from happening)
it is possible that the chunk that's provided here is (incorrectly) the one the player is currently in rather than then one they are moving into but not in yet, meaning that by waiting a tick you are probably waiting for the player to be in the chunk you intended to check.

**Comment by DelayedGaming** [comment]
@HC-Emro Can you try this [Skript nightly build](https://github.com/SkriptLang/Skript/suites/15779966227/artifacts/900551417) (from [this PR](https://github.com/SkriptLang/Skript/pull/5965)), use the chunk enter event code you have here and see if the issue has been solved?

**Comment by HC-Emro** [comment]
I just tested version 2.7.0.
My problem is with the chunk unload event, I want to retrieve the entities at the time of unload without putting ticks because this causes the server to lag when there are more than 10 players.

**Comment by HC-Emro** [comment]
![image](https://github.com/SkriptLang/Skript/assets/56135028/815fc0b5-733a-41f0-8d71-f0ef0daf7cfc)

**Comment by perlsol** [comment]
This is still an issue :(

**Comment by sovdeeth** [comment]
I can't comment on the people using skquery syntax but the original issue is actually a syntax conflict.
`[(in|of) ([world[s]] %-worlds%|1¦%-chunks%)]` means that %-worlds% will always match before %-chunks%, so the {_c} in the reported code is trying to be taken as a world, but it's a chunk, so it fails.
